<launch>

  <arg name="base_global_planner" default="global_planner/GlobalPlanner"/>
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
  
  <node name="move_base_flex" pkg="mbf_costmap_nav" type="mbf_costmap_nav" required="true" output="screen">
    
    <!-- Planner Configs -->
    <remap from="cmd_vel" to="bbot/mobile_base_controller/cmd_vel"/>
    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <rosparam file="$(find bbot_navigation)/config/planner.yaml" command="load"/>

    <!-- Load observation sources and common params for both local and global costmaps -->
    <rosparam file="$(find bbot_navigation)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find bbot_navigation)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- Load local Costmap configs -->
    <rosparam file="$(find bbot_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    
    <!-- Load global Costmap configs -->
    <rosparam file="$(find bbot_navigation)/config/costmap_global.yaml" command="load" ns="global_costmap"/>

    <!-- Load navigation configs -->
    <rosparam file="$(find bbot_navigation)/config/move_base_flex.yaml" command="load"/>

    <!-- Load recovery behaviors-->
    <rosparam file="$(find bbot_navigation)/config/recovery_behaviours.yaml" command="load"/>
  </node>

  <node name="mbf_behavior_tree" pkg="bbot_navigation" type="behavior_tree.py" output="screen"/>
</launch>